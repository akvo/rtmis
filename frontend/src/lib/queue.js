import { Store } from "pullstate";

const queue = new Store({ next: null, wait: null });

export default queue;
