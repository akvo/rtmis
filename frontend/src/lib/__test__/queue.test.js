import "@testing-library/jest-dom";
import queue from "../queue";

describe("Store", () => {
  test("check the initial state", () => {
    const state = { next: null, wait: null };
    expect(queue).toHaveProperty("initialState", state);
    expect(queue.currentState).toMatchSnapshot({
      next: null,
      wait: null,
    });
  });
});
