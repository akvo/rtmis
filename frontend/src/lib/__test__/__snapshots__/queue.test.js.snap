// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store check the initial state 1`] = `
Object {
  "next": null,
  "wait": null,
}
`;
